sanic/app.py

```diff
@@ -1179,6 +1179,8 @@ def _loop_add_task(
                 task = loop.create_task(prepped, name=name)
 
         if name and register and sys.version_info > (3, 7):
+            if isinstance(task, Task):
+                task.set_name(name)
```

Comment:
Potentially duplicative. Perhaps we should do it in an `else` block on the first conditional? Or, set an internal flag? If someone passes a task with a name already set, we would not want to override it if their intent was to just use the name for internal Sanic ref.

Reply:
When using else in the first block, it seems need to use a conditional statement that needs to check the version.
Or how about using a conditional statement in ``purge_tasks`` to first check if a name exists?

Reply:
@SaidBySolo Would you mind editing the title of the PR to reflect that you're setting the name of the task? either the semantic commit to fix(tasks) or in the rest of the descriptor. Your choice!

Reply:
@ahopkins 
How about adding an attribute like this to the Task attribute?
```py
setattr(task, "__sanic_task_name__", name)
```

Reply:
Why?

Reply:
``purge_task`` gets the registered task using task.get_name. 

If the task name and the name registered in sanic are different, a problem occurs.

Or there seems to be a workaround to not use ``get_name`` in purge_task.

Reply:
I do not think we should be adding that property unless needed to avoid renaming the task. In which case we can look for it with the `get_name` as fallback. 

Iterating over the registry probably makes sense. In this case, we would not need that hack IIRC.